dashboard/src/shared/ResourceAPIVersion.ts

```diff
@@ -0,0 +1,15 @@
+// We explicitly define the apiVersions here, just in case a generic pluralizer
+// isn't sufficient. Note that apiVersions may change over time.
```

Comment:
Couple of questions: why the new file (ie. why not define this const in `Kube.ts`).

Second, I was wondering why you were defining these when, as you say, they'll change depending on the k8s server, but then saw it'd take a bunch of queries to populate that info (eg. `/apis/rbac.authorization.k8s.io/` to get the versions, then `/apis/rbac.authorization.k8s.io/v1/` to find the available kinds and plural names, then for each api). Looks easy to hard-code :)

Checking what kubectl does, when I run `kubectl get apprepositories -v10` it is able to check cached versions of the server resources (i assume populated from the above queries).

Reply:
> Couple of questions: why the new file (ie. why not define this const in Kube.ts).

I followed the same approach than with the resource plurals that are defined as well in a standalone file (ResourceKinds.ts).

> Second, I was wondering why you were defining these when, as you say, they'll change depending on the k8s server, but then saw it'd take a bunch of queries to populate that info (eg. /apis/rbac.authorization.k8s.io/ to get the versions, then /apis/rbac.authorization.k8s.io/v1/ to find the available kinds and plural names, then for each api). Looks easy to hard-code :)

Yes, apart from the complexity of doing several requests, it would be async calls and also I am not sure how RBAC behaves in that case so it seemed better to just hardcode it for the moment.